src/analysis/plugin/plugin.py

```diff
@@ -106,34 +110,34 @@ def summarize(self, result: Schema) -> list[str]:
     def analyze(
         self,
         file_handle: io.FileIO,
-        virtual_file_path: dict,
+        virtual_file_path: VfpDict,
         analyses: dict[str, pydantic.BaseModel],
-    ) -> typing.Optional[Schema]:
+    ) -> Schema | None:
         """Analyze a file.
         May return None if nothing was found.
 
         :param file_handle: :py:class:`io.FileIO` instance of the file to be analyzed
         :param virtual_file_path: The virtual file paths, see :py:class:`~objects.file.FileObject`
         :param analyses: A dictionary of dependent analysis
 
-        :return: The analysis if anything was found.
+        :return: The analysis results (if there are any).
         """
 
-    @typing.final
-    def get_analysis(self, file_handle: io.FileIO, virtual_file_path: dict, analyses: dict[str, dict]) -> dict:
+    @final
+    def get_analysis(self, file_handle: io.FileIO, virtual_file_path: dict, analyses: dict[str, BaseModel]) -> dict:
         start_time = time.time()
         result = self.analyze(file_handle, virtual_file_path, analyses)
 
-        summary = None
-        tags = []
+        summary: list[str] | None = None
+        tags: list[Tag] = []
         if result is not None:
-            summary = self.summarize(result)
+            summary = self.summarize(result)  # type: ignore[unreachable]  # this is obviously reachable
             tags = self.get_tags(result, summary)
 
         # The dictionary as defined in the docs for FileObject.analyses_tags
         # Misses the root uid, which must be added by the scheduler
         tags_dict = {}
-        for tag in tags:
+        for tag in tags or []:
```

Comment:
Above we have `tags = []` so this is not needed, is it?

Reply:
Yes, this was a fix during the time when it was `tags = None` -> changed back